signup/tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java

```diff
@@ -591,8 +591,14 @@ private void markingRecurMeetings(List<SignupMeetingWrapper> smList) {
 	 *         current user.
 	 */
 	public boolean isAllowedToDelete() {
-		if (getSignupMeetings() == null)
+		if (sakaiFacade.isAllowedSite(sakaiFacade.getCurrentUserId(), SakaiFacade.SIGNUP_DELETE_SITE, sakaiFacade.getCurrentLocationId())) {

```

Comment:
I can see this will shortcut the loading for a someone with the permission check succeeding, but won't people without the permission still end up loading all the meetings?

Aren't we also expecting that most accesses to the tool will be by people signing up and so not have the permission on the site.